internal/querycoordv2/meta/channel_dist_manager.go

```diff
@@ -374,19 +388,27 @@ func (m *ChannelDistManager) GetShardLeader(channelName string, replica *Replica
 					case !candidatesServiceable && channelServiceable:
 						// Current candidate is not serviceable but new channel is
 						updateNeeded = true
+						logger.Debug("set updateNeeded in case 1")
```

Comment:
prefer `set serviceable delegator to candidate shard leader`  and add `zap.String("node", channel.Node)`